docs: adding markup reference docs (#883)
unoplatform/uno.extensions: doc/Reference/Markup/VisualStateManager.md

---
uid: Reference.Markup.VisualStateManager
---
# Visual State Manager

> Similar to other API's the Visual State Manager provides a fluent Builder API to help you add one or more groups each with one or more states. In it's simplest form you can add a group with a single state like:

Reviewer: ```suggestion
Similar to other API's the `VisualStateManager` provides a fluent Builder API to help adding one or more groups, each with one or more states. In its simplest form, adding a group with a single state looks like this:
```

```cs
new Grid()
	.VisualStateManager(vsm => vsm
		.Group(group => group
			.State("SomeState", state => state
				.Setters(grid => grid.Background(StaticResource.Get<Brush>("SomeResource"))))))
```

## Updating Children

> When updating the children of a given control it is important to ensure that these children exist in scope. You can provide a reference to a given control instance for the Setters to update setters specific to that control.

Reviewer: ```suggestion
When updating the children of a given control, it is important to ensure that these children exist in scope. You can provide a reference to a given control instance for the `Setters` to update those specific to the current instant.
```

```cs
new Grid()
	.Children(
		new TextBlock().Assign(out var textBlock)
	)
	.VisualStateManager(vsm => vsm
		.Group(group => group
			.State("SomeState", state => state
				.Setters(grid => grid.Background(StaticResource.Get<Brush>("SomeResource")))
				.Setters(textBlock, tb => tb.Text("You are in Some State")))));
```

## Using Triggers

> In addition to adding various style setters to a given state you may also add one or more Triggers to your Visual State.

Reviewer: ```suggestion
In addition to adding various style setters to a given state, it is also possible to add one or more `Triggers` to a Visual State.
```
